tldw_Server_API/app/api/v1/endpoints/prompt_studio/prompt_studio_websocket.py

```
@@ -215,16 +231,32 @@ async def broadcast_to_all(self, message: str):
         """Get total number of active connections."""
-        return sum(len(connections) for connections in self.active_connections.values())
+        with self._connections_lock:
+            return sum(len(connections) for connections in self.active_connections.values())
 
     def get_client_count(self) -> int:
         """Get number of unique clients connected."""
-        return len(self.active_connections)
+        with self._connections_lock:
+            return len(self.active_connections)
+
+    async def close_all(self, timeout_s: float | None = None) -> None:
```

Ruff (0.15.7)

[warning] Unused method argument: `timeout_s`

(ARG002)